TKeazirian.HTTPServer.Tests/Router/RouterTests.cs

```diff
@@ -77,4 +78,113 @@ public void ResourceCannotBeFoundReturns404()
 
         Assert.Equal("HTTP/1.1 404 Not Found\r\n", response.ResponseStatusLine);
     }
+
+    [Fact]
+    public void OptionsHandlerHasAllowHeaderWithHeadGetOptionsMethods()
+    {
+        Request testRequest = new Request("OPTIONS", "/method_options", "", "");
+        string testHeaders = "Allow: GET, HEAD, OPTIONS\r\n\r\n";
+        var testRoutesConfig = new RoutesConfig(new Dictionary<string, Handler>
+        {
+            { "/test_path", new MockHandler() },
+            { "/method_options", new SimpleOptionsHandler() }
+        });
+        Router router = new Router(testRoutesConfig);
+
+        Response response = router.Route(testRequest);
+
+        Assert.Equal("HTTP/1.1 200 OK\r\n", response.ResponseStatusLine);
+        Assert.Equal(testHeaders, response.ResponseHeaders);
+        Assert.Empty(response.GetBody());
+    }
+
+    [Fact]
+    public void OptionsHandlerHasAllowHeaderWithHeadGetOptionsPutPostMethods()
+    {
+        Request testRequest = new Request("OPTIONS", "/method_options2", "", "");
```

Comment:
in all the other tests prior to this PR, you are using mocks.  Why introduce actual request calls and couple your tests to an implementation of configuration?  Shouldn't you expand the existing mock instead?

Reply:
So would I create another Mock that looks like the actual implementation?  I think sometimes I don't do Mocks because I worry then I won't be testing the actual thing and then I won't know for sure if it's working (if that makes sense...)